### ISSUE TITLE
MaxListenersExceededWarning: Possible EventEmitter memory leak detected

### ISSUE THREAD

**Issue by ianbale** [trs/ftp-srv#37]
Hi,

I'm getting this error whenever I upload a file:

(node:49079) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 connection listeners added. Use emitter.setMaxListeners() to increase limit

The upload completes just fine, but I obiously need to address this issue.

I have this in an app that is also running Hapi, but I disabled that and I still get the same error. 

By way of a second question. is there an event that fire when a file has been uploaded?

**Comment by trs** [comment]
Re: `MaxListenersExceededWarning`
How many connections are you making to the FTP Server?
Are you adding listening events to the `FtpServer` or `Connection` instances?
Do you have other event emitters in the code you're using?

I've never seen this warning while using this module, I'm thinking more than 10 events are being listened to on either the FtpServer or Connection.


----
Re: `Event on file upload`

Current there is no event for file uploads or anything like that. That is a great idea though, I think some event overhaul would be great.

**Comment by trs** [comment]
I'm just going to lock this, I have a feeling it's just your settings for event listeners, and shouldn't be an issue.
Feel free to comment with an update.

**Comment by ianbale** [comment]
Sorry. I didn't notice your response. I commented out every other service in the app (Hapi, SFTP etc.) so that it was only this left. I was still seeing that error. 

Off hand I cannot recall whether it was the FtpServer to Connection instances I had the listeners on. Due to some other issues I ended up using another FTP module - modified to suit my very specific needs (a client with a very strange FTP client sending us data).
